### PR TITLE
Develop.sprint8.chat sample

### PR DESCRIPTION
**Pull request by IegorKozakov**
Auto login
Updating dialog
Notification messages

### CODE REVIEW

File: samples/newChat/css/dashboard.css [hunk 1]
```diff
@@ -538,12 +302,22 @@
     margin: 0 0 16px;
 }
 
+.message__wrap.m-notification_message {
+    display: block;
+    color: #778594;
+    font-family: Roboto;
```
Comment (dimaspirit) [inline samples/newChat/css/dashboard.css]: Why do you set font-family? If you set Roboto to body .

File: samples/newChat/css/dashboard.css [hunk 2]
```diff
@@ -538,12 +302,22 @@
...
+    font-size: 13px;
+    font-weight: 500;
```
Comment (dimaspirit) [inline samples/newChat/css/dashboard.css]: use font: 500 13px/15px Roboto, sans-serif;
if you set all of property

File: samples/newChat/css/dashboard.css [hunk 3]
```diff
@@ -705,14 +479,76 @@
...
+.group_chat__title,
+.update_chat__title_input {
     font-size: 22px;
     line-height: 25px;
     color: #4A4A4A;
     flex: 1 1 100%;
 }
 
-.group_chat__user_list {
+.update_update_chat_name_form {
```
Comment (dimaspirit) [inline samples/newChat/css/dashboard.css]: update_update - seriously?

File: samples/newChat/css/dashboard.css [hunk 4]
```diff
@@ -705,14 +479,76 @@
...
+.update_update_chat_name_form.active i{
+    color: #18C942;
+}
+
+.update_chat__title_button {
+    border: none;
+    background: #fff;
+}
+
+
```
Comment (dimaspirit) [inline samples/newChat/css/dashboard.css]: remove line

File: samples/newChat/css/dashboard.css [hunk 5]
```diff
@@ -768,13 +614,18 @@
     color: #778594;
 }
 
-.create_dialog_form {
+.create_dialog_form,
+.update_dialog_form {
```
Comment (dimaspirit) [inline samples/newChat/css/dashboard.css]: Are you can use one selector with modifier for form if you form has equals style.

File: samples/newChat/css/dashboard.css [hunk 6]
```diff
@@ -788,27 +639,37 @@
     cursor: no-drop;
 }
 
-.create_dialog_btn {
-    width: 96px;
-    flex: 0 0 96px;
+.create_dialog_btn,
```
Comment (dimaspirit) [inline samples/newChat/css/dashboard.css]: set unified class ```.btn``` for buttons, so many repeats

File: samples/newChat/css/dialogs.css [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+    background-color: #54C7FC;
+}
+
+.dialog__avatar.m-type_group.m-user__img_10 {
+    color: #b3bbe0;
+    background-color: #4054B2;
+}
+
+.message__avatar.m-user__img_not_loaded,
+.message__avatar.m-typing_uncnown {
```
Comment (dimaspirit) [inline samples/newChat/css/dialogs.css]: uncnown => unknown

File: samples/newChat/css/login.css [hunk 8]
```diff
@@ -44,38 +44,47 @@
...
-    transform: translate(0, -50%);
-    border-left: 5px solid transparent;
-	border-right: 5px solid transparent;
-	border-top: 5px solid #778594;
+    left: 0;
+    top: 20px;
+    font-size: 16px;
+    line-height: 19px;
+    color: #C5CACE;
+    transition: 0.15s;
```
Comment (dimaspirit) [inline samples/newChat/css/login.css]: better use all properties: what will animate, time, func of animation

File: samples/newChat/index.html [hunk 9]
```diff
@@ -28,21 +28,28 @@
...
-                    <div class="login__button_wrap">
-                        <button class="login__button j-login__button" disabled>
-                            login
-                        </button>
-                    </div>
+                    
+                    <form name="loginForm" class="login__form">
+                        <div class="login_form__row">
+                            <label for="userName">User name</label>
+                            <input type="text" id="userName" name="userName"/>
```
Comment (dimaspirit) [inline samples/newChat/index.html:40]: Could you add to spec patterns to input data as username, usergroup

File: samples/newChat/css/dashboard.css (hunk 1 above)
Comment (Vladlukhanin) [reply samples/newChat/css/dashboard.css]: And may be need alternative fonts - font-family: Roboto, sans-serif;

File: samples/newChat/js/app.js [hunk 10]
```diff
@@ -27,93 +29,46 @@ App.prototype.init = function (config) {
...
-        }
-    });
-};
 
-App.prototype.loadDashboard = function () {
-    var self = this,
-        logoutBtn = document.querySelector('.j-logout');
-    
+    if(activeTabName){
+        renderParams.tabName = activeTabName
```
Comment (dimaspirit) [inline samples/newChat/js/app.js]: ```;``` is all what we need

File: samples/newChat/js/listeners.js [hunk 11]
```diff
@@ -48,20 +51,61 @@ Listeners.prototype.onMessageListener = function (userId, message) {
...
+            dialog.users.push(userId);
+        } else if(extension.dialog_name){
+            dialog.name = extension.dialog_name;
+            dialogModule.updateDialogUi(dialogId, extension.dialog_name);
+        }
+    }
+
+    var activeTab = document.querySelector('.j-sidebar__tab_link.active'),
+        tabType = activeTab.dataset.type,
+        dialogType = dialog.type === 1 ? "public" : "chat";
```
Comment (dimaspirit) [inline samples/newChat/js/listeners.js]: use singlequote

File: samples/newChat/js/listeners.js [hunk 12]
```diff
@@ -85,18 +129,37 @@ Listeners.prototype.onReadStatusListener = function () {
 };
 
 Listeners.prototype.onSystemMessageListener = function (message) {
-    if (message.extension && (message.extension.notification_type === '1' || message.extension.notification_type === 'creating_dialog')) {
+    var dialog = dialogModule._cache[message.dialog_id || message.extension.dialog_id];
+
+    if (message.extension && message.extension.notification_type === '1') {
```
Comment (dimaspirit) [inline samples/newChat/js/listeners.js]: set  '1' as CONST

File: samples/newChat/js/listeners.js [hunk 13]
```diff
@@ -85,18 +129,37 @@ Listeners.prototype.onReadStatusListener = function () {
...
+                dialogModule._cache[dialog._id] = helpers.compileDialogParams(dialog);
                 var type = dialog.type === 1 ? 'public' : 'chat',
                     activeTab = document.querySelector('.j-sidebar__tab_link.active');
 
                 if (activeTab && type === activeTab.dataset.type) {
-                    dialogModule.renderDialog(dialog, true);
+                    dialogModule.renderDialog(dialogModule._cache[dialog._id], true);
                 }
+            }).catch(function(error){
+                console.error(error);
```
Comment (dimaspirit) [inline samples/newChat/js/listeners.js:145]: Is it all handle if we don't found the dialog?

File: samples/newChat/js/message.js [hunk 14]
```diff
@@ -3,7 +3,7 @@
 function Message() {
     this.container = null;
     this.attachmentPreviewContainer = null;
-    this.typingTymeout = appConfig.typingTymeout || 3
+    this.typingTymeout = appConfig.typingTymeout || 3;
```
Comment (dimaspirit) [inline samples/newChat/js/message.js]: typingTymeout => typingTimeout

File: samples/webrtc/js/helpers.js [hunk 15]
```diff
@@ -156,6 +156,8 @@
                     reject(csErr);
                 } else {
                     /** In first trying to login */
+
+                    console.log(userRequiredParams);
```
Comment (dimaspirit) [inline samples/webrtc/js/helpers.js]: remove console.log

**Pull request merged**